### PR TITLE
Role selection

### PR DESCRIPTION
**Pull request by RanjithaTK**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,26 +1,28 @@
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from "@angular/core"
```
Comment (vishwanath1004) [inline src/app/app.module.ts:1]: No changes i this file apart from single  quotes to double qoutes.

File: src/app/modules/auth/role-selection/role-selection.component.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<div class="content">
+  <div class="card-wrapper">
+    <mat-card
```
Comment (vishwanath1004) [inline src/app/modules/auth/role-selection/role-selection.component.html:3]: Make this as component

File: src/app/shared/components/role-selection-card/role-selection-card.component.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  selectedRole = ""
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  cardSelected(role: any) {
+    this.selectedRole = role.role
+    this.cardOneSelected.emit(this.selectedRole)
+    console.log(this.selectedRole)
```
Comment (vishwanath1004) [inline src/app/shared/components/role-selection-card/role-selection-card.component.ts:20]: remove console logs

File: src/styles.scss [hunk 4]
```diff
@@ -37,3 +37,6 @@ $mentorEd-portal-theme: mat.define-light-theme((
 
 html, body { height: 100%; }
 body { margin: 0; font-family: Roboto, "Helvetica Neue", sans-serif; }
+
```
Comment (vishwanath1004) [inline src/styles.scss:40]: remove this duplicate

File: src/app/shared/components/role-selection-card/role-selection-card.component.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Component, Input, OnInit, Output, EventEmitter } from "@angular/core"
+// import * from "../../../../assets/images/role-selection/"
```
Comment (vishwanath1004) [inline src/app/shared/components/role-selection-card/role-selection-card.component.ts:2]: remove unwanted or commented code

File: angular.json [hunk 6]
```diff
@@ -97,6 +98,7 @@
               "src/assets"
             ],
             "styles": [
+              "./node_modules/@angular/material/prebuilt-themes/indigo-pink.css",
```
Comment (kiranharidas187) [inline angular.json:101]: Why are we doing these theme changes??

File: src/app/app.module.ts [hunk 7]
```diff
@@ -1,12 +1,12 @@
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core'
```
Comment (kiranharidas187) [inline src/app/app.module.ts:1]: Please remove all unwanted changes in this page

File: src/app/modules/auth-routing.module.ts [hunk 8]
```diff
@@ -1,39 +1,42 @@
-import { NgModule } from "@angular/core";
-import { RouterModule, Routes } from '@angular/router';
-import { AuthComponent } from "./auth/auth.component";
-import { LoginComponent } from "./auth/login/login.component";
-import { RegisterComponent } from "./auth/register/register.component";
+import { NgModule } from "@angular/core"
```
Comment (kiranharidas187) [inline src/app/modules/auth-routing.module.ts:1]: Please remove these unwanted changes

File: src/app/modules/auth/role-selection/role-selection.component.scss [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+.content {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+}
+
+.button {
+  width: 20rem;
+  background: rgb(105, 104, 104);
```
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/modules/auth/role-selection/role-selection.component.scss:8]: is this required now?

File: src/app/modules/auth/role-selection/role-selection.component.scss [hunk 10]
```diff
@@ -0,0 +1,11 @@
+
+.content {
+  padding-top: 15%;
+}
+
+.button {
+  width: 100%;
+  background: rgb(105, 104, 104);
+  color: white;
```
Comment (kiranharidas187) [inline src/app/modules/auth/role-selection/role-selection.component.scss:9]: use color variable

File: src/app/shared/components/role-selection-card/role-selection-card.component.scss [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+.heading {
+  display: flex;
+  justify-content: center;
+  padding-top: 8%;
+  font-weight: 600;
+  color: #000;
+}
+.sub-content {
+  color: #000;
```
Comment (kiranharidas187) [inline src/app/shared/components/role-selection-card/role-selection-card.component.scss:42]: use color variables

File: src/app/shared/components/role-selection-card/role-selection-card.component.scss [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+
+
+.heading {
+  display: flex;
+  justify-content: center;
+  padding-top: 8%;
+  font-weight: 600;
+  color: #000;
```
Comment (kiranharidas187) [inline src/app/shared/components/role-selection-card/role-selection-card.component.scss:39]: use color variables

File: src/app/shared/components/role-selection-card/role-selection-card.component.scss [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+.mentor-image {
+  width: 7rem;
+  height: 7rem;
+}
+
+
+
+.heading {
+  display: flex;
+  justify-content: center;
```
Comment (kiranharidas187) [inline src/app/shared/components/role-selection-card/role-selection-card.component.scss:36]: can we use fx classes for 35 and 36th line css

File: src/app/shared/components/role-selection-card/role-selection-card.component.scss [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+
+.mentor-card-selected {
+  background-color: mat-color($sb-palette-primary,400) !important;
+  .card-details {
+    color: mat-color($sb-palette-primary, 50) !important;
+  }
+}
+mat-card-header {
+  align-items: center;
+  justify-content: center;
```
Comment (kiranharidas187) [inline src/app/shared/components/role-selection-card/role-selection-card.component.scss:24]: can we use fx classes for 23 and 24th line css

File: src/app/modules/auth/role-selection/role-selection.component.scss [hunk 15]
```diff
@@ -0,0 +1,14 @@
+@import "../../../../sb-theme.scss";
```
Comment (kiranharidas187) [inline src/app/modules/auth/role-selection/role-selection.component.scss:1]: rm if this is not used

File: src/app/shared/components/role-selection-card/role-selection-card.component.scss [hunk 16]
```diff
@@ -0,0 +1,37 @@
+
+@import '../../../../sb-theme.scss';
```
Comment (kiranharidas187) [inline src/app/shared/components/role-selection-card/role-selection-card.component.scss:2]: rm if not used

**Pull request merged**